compiler/src/test/java/com/squareup/anvil/compiler/dagger/BindsMethodValidatorTest.kt

```diff
@@ -164,14 +194,56 @@ class BindsMethodValidatorTest(
     }
   }
 
+  @Test
+  fun `an extension function binding is valid`() {
+    val moduleResult = compile(
+      """
+      package com.squareup.test
+ 
+      import dagger.Binds
+      import dagger.Module
+      import javax.inject.Inject
+
+      class Foo @Inject constructor() : Bar
+      interface Bar
+
+      @Module
+      abstract class BarModule {
+        @Binds
+        abstract fun Foo.bindsBar(): Bar
+      }
+      """
+    ) {
+      assertThat(exitCode).isEqualTo(OK)
+    }
+
+    compile(
+      """
+      package com.squareup.test
+
+      import dagger.Component
+      import javax.inject.Singleton
+
+      @Component(modules = [BarModule::class])
+      @Singleton
+      interface ComponentInterface
```

Comment:
```suggestion
      interface ComponentInterface {
        fun bar(): Bar
      }
```
Otherwise Dagger won't care about the binding (or in different words, you could remove the Dagger module from the component and it still would compile).

Reply:
Good catch 👍